router/router.go

```diff
@@ -0,0 +1,340 @@
+	}
+	return nil
+}
+
+func (ctx *Router) load() error {
+	ctx.mutexScan.Lock()
+	defer ctx.mutexScan.Unlock()
+	log.Printf("Loading alerts configuration file %s ....\n", ctx.cfgfile)
+	tenant, err := Parsev2cfg(ctx.cfgfile)
+
+	if err != nil {
+		return err
+	}
+
+	if len(tenant.AquaServer) > 0 {
+		var slash string
+		if !strings.HasSuffix(tenant.AquaServer, "/") {
+			slash = "/"
+		}
+		ctx.aquaServer = fmt.Sprintf("%s%s#/images/", tenant.AquaServer, slash)
+	}
+
+	dbservice.DbSizeLimit = tenant.DBMaxSize
+	dbservice.DbDueDate = tenant.DBRemoveOldData
+	if tenant.DBTestInterval == 0 {
+		tenant.DBTestInterval = 1
+	}
+	if dbservice.DbSizeLimit != 0 || dbservice.DbDueDate != 0 {
+		ctx.ticker = time.NewTicker(baseForTicker * time.Duration(tenant.DBTestInterval))
+		go func() {
+			for {
+				select {
+				case <-ctx.stopTicker:
+					return
+				case <-ctx.ticker.C:
+					dbservice.CheckSizeLimit()
+					dbservice.CheckExpiredData()
+				}
+			}
+		}()
+	}
+
+	for i, r := range tenant.InputRoutes {
+		ctx.inputRoutes[r.Name] = routes.ConfigureAggrTimeout(&tenant.InputRoutes[i])
+	}
+	for _, t := range tenant.Templates {
+		err := ctx.initTemplate(&t)
+		if err != nil {
+			log.Printf("Can not initialize template %s: %v \n", t.Name, err)
+		}
+	}
+
+	for _, settings := range tenant.Outputs {
+		utils.Debug("%#v\n", anonymizeSettings(&settings))
+
+		if settings.Enable {
+			plg := BuildAndInitOtpt(&settings, ctx.aquaServer)
+			if plg != nil {
+				log.Printf("Output %s is configured", settings.Name)
+				ctx.outputs[settings.Name] = plg
+			}
+		}
+	}
+	return nil
+}
+
+type service interface {
+	MsgHandling(input []byte, output outputs.Output, route *routes.InputRoute, inpteval data.Inpteval, aquaServer *string)
+}
+
+var getScanService = func() service {
+	serv := &msgservice.MsgService{}
+	return serv
+}
+var getHttpClient = func() *http.Client {
+	return http.DefaultClient
+}
+
+func (ctx *Router) HandleRoute(routeName string, in []byte) {
+	r, ok := ctx.inputRoutes[routeName]
+	if !ok || r == nil {
+		log.Printf("There isn't route %q", routeName)
+		return
+	}
+	if len(r.Outputs) == 0 {
+		log.Printf("route %q has no outputs", routeName)
+		return
+	}
+	for _, outputName := range r.Outputs {
+		pl, ok := ctx.outputs[outputName]
+		if !ok {
+			log.Printf("route %q contains an output %q, which doesn't enable now.", routeName, outputName)
+			continue
+		}
+		tmpl, ok := ctx.templates[r.Template]
+		if !ok {
+			log.Printf("route %q contains reference to undefined or misconfigured template %q.",
+				routeName, r.Template)
+			continue
+		}
+		log.Printf("route %q is associated with template %q", routeName, r.Template)
+		go getScanService().MsgHandling(in, pl, r, tmpl, &ctx.aquaServer)
+	}
+}
+
+func (ctx *Router) handle(in []byte) {
+	for routeName := range ctx.inputRoutes {
+		ctx.HandleRoute(routeName, in)
+	}
+}
+func BuildAndInitOtpt(settings *OutputSettings, aquaServerUrl string) outputs.Output {
+	var plg outputs.Output
```

Comment:
I'd transfer to line 305 (before `switch` and using).

Reply:
- [ ] move variable definition before line `switch` statement

Reply:
done